src/snowflake/snowpark/modin/plugin/compiler/snowflake_query_compiler.py

```diff
@@ -9718,6 +9718,17 @@ def resample(
                     )
                     .set_index_names([None])
                 )
+            elif resample_method in ("first", "last"):
+                # Call groupby_first or groupby_last directly
+                qc = getattr(
+                    SnowflakeQueryCompiler(frame), "groupby_" + resample_method
```

Comment:
f-strings are preferred to `+`
```suggestion
                    SnowflakeQueryCompiler(frame), f"groupby_{resample_method}"
```